test/extensions/key_value/file_based/key_value_store_test.cc

```diff
@@ -34,42 +37,74 @@ class KeyValueStoreTest : public testing::Test {
   std::string filename_;
   std::unique_ptr<FileBasedKeyValueStore> store_{};
   std::chrono::seconds flush_interval_{5};
+  Event::MockTimer* ttl_timer_ = nullptr;
   Event::MockTimer* flush_timer_ = nullptr;
+  Event::SimulatedTimeSystem test_time_;
 };
 
 TEST_F(KeyValueStoreTest, Basic) {
   EXPECT_EQ(absl::nullopt, store_->get("foo"));
-  store_->addOrUpdate("foo", "bar");
+  store_->addOrUpdate("foo", "bar", absl::nullopt);
   EXPECT_EQ("bar", store_->get("foo").value());
-  store_->addOrUpdate("foo", "eep");
+  store_->addOrUpdate("foo", "eep", absl::nullopt);
   EXPECT_EQ("eep", store_->get("foo").value());
   store_->remove("foo");
   EXPECT_EQ(absl::nullopt, store_->get("foo"));
 }
 
+TEST_F(KeyValueStoreTest, TTL) {
+  test_time_.setSystemTime(std::chrono::milliseconds(0));
+  EXPECT_CALL(*ttl_timer_, enableTimer(std::chrono::milliseconds(5000), _));
+  store_->addOrUpdate("foo", "bar", std::chrono::seconds(5));
+  EXPECT_EQ("bar", store_->get("foo").value());
+  // Advance timer to trigger expiry
+  test_time_.setSystemTime(std::chrono::milliseconds(10000));
```

Comment:
nit: I would expect the timer to go off at the time it is scheduled to on line 57, namely after 5s not after 10s.

Reply:
Maybe this comment is based on an old revision? or I'm missing something

Reply:
Here is the code I see in the current version of the PR:

```
  test_time_.setSystemTime(std::chrono::milliseconds(0));
  EXPECT_CALL(*ttl_timer_, enableTimer(std::chrono::milliseconds(5000), _));
  store_->addOrUpdate("foo", "bar", std::chrono::seconds(5));
  EXPECT_EQ("bar", store_->get("foo").value());
  // Advance timer to trigger expiry
  test_time_.setSystemTime(std::chrono::milliseconds(10000));
  ttl_timer_->invokeCallback();
```

The callback seems to be invoked when the time is 10s. (The timer is *enabled* to fire when the time is 5s but that's not when the callback is invoked.)

And actually, we might want to tweak the comment from "Advance timer to trigger expiry" to something like "Advance time to when the timer is scheduled to fire".

Reply:
Oh I understand now. I was mixing up expect and `EXPECT`, if that makes sense.  I can change to the precise numbers throughout.  

And sure can do